sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/IntervalUtils.scala

```diff
@@ -181,8 +181,35 @@ object IntervalUtils {
               "year-month", YM(startField, endField).typeName)
           }
         }
-      case _ => throwIllegalIntervalFormatException(input, startField, endField,
-        "year-month", YM(startField, endField).typeName)
+      case _ =>
+        try {
+          val calendar = stringToInterval(input)
```

Comment:
Why did you implement parsing of non-ANSI interval in the function which is supposed to parse ANSI year-month interval strings?